Form/FormFlow.php

```diff
@@ -609,6 +645,16 @@ protected function bindFlow() {
 			$reset = true;
 		}
 
+		if ($this->getRequest()->isMethod('POST') && ! $this->dataManager->hasInstance($this)) {
+			// Form is stale, kill posted data and reset
+			$this->getRequest()->request->replace();
+			$reset = true;
+			$this->setStale(true);
+
+                        // Regenerate instance ID
+			$this->setInstanceId($this->determineInstanceId());
```

Comment:
Generating a new instance id doesn't work in "redirect after submit" mode. What's the reason for a new id anyway? The form is stale if no data is present (anymore), so the previous id can just be reused.